api/eds.proto

```diff
@@ -47,48 +47,37 @@ message EndpointDiscoveryRequest {
   Node node = 2;
 }
 
-enum HealthStatus {
-  UNKNOWN = 0;
-  HEALTHY = 1;
-  UNHEALTHY = 2;
-  // Connection draining in progress -
-  // https://aws.amazon.com/blogs/aws/elb-connection-draining-remove-instances-from-service-with-care/
-  // and
-  // https://cloud.google.com/compute/docs/load-balancing/enabling-connection-draining.
-  DRAINING = 3;
-  // This value is used by HDS Remote server. From Envoy’s perspective
-  // TIMEOUT = UNHEALTHY in case EDS returns HealthStatus.
-  TIMEOUT = 4;
-}
-
 message LbEndpoint {
-  // Endpoint host address.
-  Address address = 1;
-  // Optional health status when known and supplied by EDS server.
-  HealthStatus health_status = 2;
+  EndpointHealth endpoint_health = 1;
   // The optional canary status of the upstream host. Envoy uses the canary
   // status for various statistics and load balancing tasks documented
   // elsewhere.
-  google.protobuf.BoolValue canary = 3;
+  google.protobuf.BoolValue canary = 2;
   // The optional load balancing weight of the upstream host, in the range 1 -
   // 100. Envoy uses the load balancing weight in some of the built in load
-  // balancers.
-  // TODO(htuch): How does this combine with the LocalityEndpoints weight?
-  google.protobuf.UInt32Value load_balancing_weight = 4;
+  // balancers. All load balancing weights in a locality must sum to 100%.
+  // The effective load balancing weight is load_balancing_weight multiplied by
+  // the locality wide load_balancing_weight. If unspecified, each host is
+  // presumed to have equal weight in a locality.
+  google.protobuf.UInt32Value load_balancing_weight = 3;
 }
 
 // All endpoints belonging to a Locality.
 message LocalityEndpoints {
   Locality locality = 1;
   repeated LbEndpoint lb_endpoints = 2;
 
-  // Optional: Per region/zone/sub_zone weight - range 1-100.
-  // TODO(htuch): Is there some restriction that all weights in a cluster must
-  // sum to 100? Do we want to be more specific and have per-zone and
-  // per-subzone as distinct weights for example? Can the weight also be 0?
+  // Optional: Per region/zone/sub_zone weight - range 1-100. All load balancing
+  // weights in a cluster must sum to 100%. If unspecified, each locality is
+  // presumed to have equal weight in a cluster.
   google.protobuf.UInt32Value load_balancing_weight = 3;
 }
 
+// Each route from RDS will map to a single cluster (or specify its own weights
+// for WeightedCluster which are discoverable by RDS). In EDS, each cluster is
```

Comment:
Done, I will also add some stuff to a `TERMINOLOGY.md` file soon to help clarify.